lib/ecmascript.ts

```diff
@@ -872,7 +863,22 @@ export function ToSecondsStringPrecision(options: Temporal.ToStringPrecisionOpti
   unit: UnitSmallerThanOrEqualTo<'minute'>;
   increment: number;
 } {
-  const smallestUnit = ToSmallestTemporalUnit(options, undefined, ['year', 'month', 'week', 'day', 'hour']);
+  const smallestUnit = GetTemporalUnit(options, 'smallestUnit', 'time', undefined);
+  // This condition is impossible according to type-checking, so we have to
+  // widen the return from GetTemporalUnit to trick tsc.
+  // This could potentially be gated behind DEBUG to remove unnecessary code iff
+  // users are comfortable relying on the typechecker instead.
+  if ((smallestUnit as Temporal.TimeUnit) === 'hour') {
```

Comment:
Not sure I get why this is necessary from just reading the comment and looking at the signature of the first overload of GetTemporalUnit. I'd expect the return type of that overload to be `Temporal.TimeUnit | 'auto' | undefined` which seems like it shouldn't make the condition impossible.

Reply:
This is leftover from a previous iteration of the GetTemporalUnit overloads which I screwed up. Thanks for catching - you are right about the return type and this not needing the cast.